extension/storage/filestorage/config.go

```diff
@@ -29,24 +30,44 @@ type Config struct {
 
 	Directory string        `mapstructure:"directory,omitempty"`
 	Timeout   time.Duration `mapstructure:"timeout,omitempty"`
+
+	Compaction *CompactionConfig `mapstructure:"compaction,omitempty"`
+}
+
+type CompactionConfig struct {
+	OnStart            bool   `mapstructure:"on_start,omitempty"`
+	Directory          string `mapstructure:"directory,omitempty"`
+	MaxTransactionSize int64  `mapstructure:"max_transaction_size,omitempty"`
 }
 
 func (cfg *Config) Validate() error {
-	info, err := os.Stat(cfg.Directory)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return fmt.Errorf("directory must exist: %v", err)
+	var dirs []string
+	if cfg.Compaction.OnStart {
+		dirs = []string{cfg.Directory, cfg.Compaction.Directory}
+	} else {
+		dirs = []string{cfg.Directory}
+	}
+	for _, dir := range dirs {
+		info, err := os.Stat(dir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return fmt.Errorf("directory must exist: %v", err)
+			}
+			if fsErr, ok := err.(*fs.PathError); ok {
+				return fmt.Errorf(
+					"problem accessing configured directory: %s, err: %v",
+					dir, fsErr,
+				)
+			}
+
 		}
-		if fsErr, ok := err.(*fs.PathError); ok {
-			return fmt.Errorf(
-				"problem accessing configured directory: %s, err: %v",
-				cfg.Directory, fsErr,
-			)
+		if !info.IsDir() {
+			return fmt.Errorf("%s is not a directory", dir)
 		}
-
 	}
-	if !info.IsDir() {
-		return fmt.Errorf("%s is not a directory", cfg.Directory)
+
+	if cfg.Compaction.MaxTransactionSize < 0 {
```

Comment:
Can we use default value? Is 0 a valid value?

Reply:
A value of zero will ignore transaction sizes
https://github.com/etcd-io/bbolt/blob/master/compact.go

Reply:
I had in mind a default non-zero transaction size (if it was not set). But I believe zero should be fine as default